cedar-policy-validator/src/human_schema/to_json_schema.rs

```diff
@@ -46,12 +46,20 @@ use super::{
 /// This will let you subsequently decode that into the Validator AST for Schemas ([`crate::ValidatorSchema`]).
 /// On success, this function returns a tuple containing:
 ///     * The `SchemaFragment`
-///     * A vector of name collisions, that are essentially warnings
+///     * An iterator of warnings that were generated
+///
+/// TODO: In the future, these warnings should be generated later in the
+/// process, such that we apply the same checks to JSON and human schemas
 pub fn custom_schema_to_json_schema(
     schema: Schema,
     extensions: Extensions<'_>,
 ) -> Result<(SchemaFragment<RawName>, impl Iterator<Item = SchemaWarning>), ToJsonSchemaErrors> {
-    // First pass, figure out what each name is bound to
+    // combine all of the declarations in unqualified (empty) namespaces into a
+    // single unqualified namespace
+    //
+    // TODO: In the future, we could also combine namespaces with matching
+    // non-empty names, to allow reopening a namespace within the same (human)
+    // schema fragment.
```

Comment:
The comment implies that we allow it in the JSON, but not human-readable syntax? That seems worth filing an issue for.

Reply:
We can't allow reopening a namespace within the same JSON schema fragment, because the JSON schema structure is a map from namespace name to declarations, so it would be a duplicate key in the JSON map.  However, we could allow it in the human-readable syntax.

Reply:
reopening a namespace between fragments should work.

Reply:
Opened #1086 for this, and pointed this TODO at that issue